python/doc/examples/meta_modeling/polynomial_chaos_metamodel/plot_pce_design.py

```diff
@@ -0,0 +1,447 @@
+# .. math::
+#
+#     H := D \left(D^T D\right)^{-1} D^T.
+#
+# The hat matrix puts a hat to the vector of observations to
+# produce the vector of predictions of the linear model:
+#
+# .. math::
+#
+#    \hat{\vect{y}} = H \vect{y}
+#
+# To solve a linear least squares problem, we need to evaluate the
+# design matrix :math:`D`, which is the primary goal of
+# the :class:`~openturns.DesignProxy` class.
+# Let us present some examples of situations where the design matrix
+# is required.
+#
+# - When we use the QR decomposition, we actually do not need to evaluate it in
+#   our script: the :class:`~openturns.QRMethod` class knows how to compute the
+#   solution without evaluating the Gram matrix `D^T D`.
+# - We may need the inverse Gram matrix
+#   :math:`(D^T D)^{-1}` sometimes, for example when we want to create
+#   a D-optimal design.
+# - Finally, when we want to compute the analytical leave-one-out error,
+#   we need to compute the diagonal of the  projection matrix :math:`H`.
+#
+# For all these purposes, the `DesignProxy` is *the* tool.
+
+# %%
+# The leave-one-out error
+# -----------------------
+# In this section, we show that the leave-one-error of a regression problem
+# can be computed using an analytical formula which depends on the hat matrix
+# :math:`H`.
+# We consider the physical model:
+#
+# .. math::
+#
+#     y = g(\vect{x})
+#
+# where :math:`\vect{x} \in \Rset^{n_X}` is the input and :math:`y \in \Rset` is
+# the output.
+# Consider the problem of approximating the physical model :math:`g` by the
+# linear model:
+#
+# .. math::
+#
+#     \hat{y} := \tilde{g}(\vect{x}) = \sum_{k = 1}^m a_k \psi_k(\vect{x})
+#
+# for any :math:`\vect{x} \in \Rset^{n_X}` where :math:`\{\psi_k : \Rset^{n_X} \rightarrow \Rset\}_{k = 1, \ldots, m}` are the basis functions and
+# :math:`\vect{a} \in \Rset^m` is a vector of parameters.
+# The mean squared error is ([blatman2009]_ eq. 4.30 page 85):
+#
+# .. math::
+#
+#     \operatorname{MSE}\left(\tilde{g}\right)
+#     = \mathbb{E}_{\vect{X}}\left[\left(g\left(\vect{X}\right) - \tilde{g}\left(\vect{X}\right) \right)^2 \right]
+#
+# The leave-one-out error is an estimator of the mean squared error.
+# Let:
+#
+# .. math::
+#
+#    \cD = \{\vect{x}^{(1)}, \ldots, \vect{x}^{(n)} \in \Rset^{n_X}\}
+#
+# be independent observations of the input random vector :math:`\vect{X}` and
+# let :math:`\{y^{(1)}, \ldots, y^{(n)} \in \Rset^{n_X}\}` be the corresponding
+# observations of the output of the physical model:
+#
+# .. math::
+#
+#     y^{(i)} = g\left(\vect{x}^{(i)}\right)
+#
+# for :math:`i = 1, ..., n`.
+# Let :math:`\vect{y} \in \Rset^n` be the vector of observations:
+#
+# .. math::
+#
+#     \vect{y} = (y^{(1)}, \ldots, y^{(n)})^T.
+#
+#
+# Consider the following set of inputs, let aside the :math:`i`-th input:
+#
+# .. math::
+#
+#     \cD_{-i} := \left\{\vect{x}^{(1)}, \ldots, \vect{x}^{(i - 1)}, \vect{x}^{(i + 1)}, \ldots, \vect{x}^{(n)}\right\}
+#
+# for :math:`i \in \{1, ..., n\}`.
```

Comment:
The letter $i$ should be changed for $j$, so that the notations can be kept consistent across pages.

Reply:
Fixed at b6fa07671d806cb4189983595d4f50028b5d664a